packages/react-bindings/src/hooks/useStyles.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  ComponentSlotStyle,
+  ComponentVariablesInput,
+  DebugData,
+  emptyTheme,
+} from '@fluentui/styles'
+import * as React from 'react'
+// @ts-ignore We have this export in package, but it is not present in typings
+import { ThemeContext } from 'react-fela'
+
+import { ComponentDesignProp, RendererRenderRule, StylesContextValue } from '../styles/types'
+import getStyles, { GetStylesResult } from '../styles/getStyles'
+
+type PrimitiveProps = Record<string, boolean | number | string | undefined>
+type UseStylesOptions<StyleProps extends PrimitiveProps> = {
+  className?: string
+  mapPropsToStyles?: () => StyleProps
+  mapPropsToInlineStyles?: () => InlineStyleProps<StyleProps>
+  rtl?: boolean
+}
+
+type InlineStyleProps<StyleProps> = {
+  /** Additional CSS class name(s) to apply.  */
+  className?: string
+
+  design?: ComponentDesignProp
+
+  /** Additional CSS styles to apply to the component instance.  */
+  styles?: ComponentSlotStyle<StyleProps, any> // TODO: see if we can improve it
+
+  /** Override for theme site variables to allow modifications of component styling via themes. */
+  variables?: ComponentVariablesInput
+}
+
+const defaultContext: StylesContextValue<{ renderRule: RendererRenderRule }> = {
+  disableAnimations: false,
+  renderer: { renderRule: () => '' },
+  theme: emptyTheme,
+  _internal_resolvedComponentVariables: {},
+}
+
+const useStyles = <StyleProps extends PrimitiveProps>(
+  displayName: string,
+  options: UseStylesOptions<StyleProps>,
+): [GetStylesResult['classes'], GetStylesResult['styles']] => {
+  const context: StylesContextValue<{ renderRule: RendererRenderRule }> =
+    React.useContext(ThemeContext) || defaultContext
```

Comment:
Any ideas on how we could decouple this call from the concrete theme? It would be nice to be able to reuse the base components with a different design system (which would likely have a separate theme shape).

Reply:
This is next stage goal 🙇‍♂️

This is actually even not `ThemeContext` as it contains everything, like `renderer` and `disableAnimations`. Currently we can't we have multiple contexts as the most of our components are class components 🙃

I have a naive idea that we will be able to move out `renderer` to a separate context and use the same interface different engines, https://codesandbox.io/embed/t-rex-proto-1w8bp